File: static/code/sample/tethered/gdl-semotion-b.js
// In this sample:
// Detect motion

import pkg_serialusb from 'dlserialusb';
const {SerialUSB} = pkg_serialusb

import pkg_duelink from 'duelink';
const {DUELinkController} = pkg_duelink

let duelink = new DUELinkController(new SerialUSB());
await duelink.Connect();

// Sleep helper (ms)
function sleep(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
}

// Methods
async function Detect() {
    return await duelink.Engine.ExecuteCommand("Detect()");
}

(async () => {
    while (true) {
        const detect = await Detect();

        if (detect !== 0) {
            console.log("Motion Detected");
        }

        await sleep(1000); // sleep 1 second
    }
})();

File: static/code/sample/tethered/gdl-semotion-b.py
# In this sample:
# Detect motion

import time
from DUELink.DUELinkController import DUELinkController

availablePort = DUELinkController.GetConnectionPort()
duelink = DUELinkController(availablePort)

# Methods
def Detect():
    return int(duelink.Engine.ExecuteCommand("Detect()"))

while True:
    detect = Detect()

    if detect != 0:
        print("Motion Detected")

    time.sleep(1)  # sleep for 1 second